application/controllers/collection_instrument.py

```diff
@@ -277,17 +278,43 @@ def unlink_instrument_from_exercise(self, instrument_id, exercise_id, session=No
             bound_logger.info("Failed to unlink, unable to find instrument or exercise")
             raise RasError("Unable to find instrument or exercise", 404)
 
-        instrument.exercises.remove(exercise)
-        for business in instrument.businesses:
-            bound_logger.info("Removing business/exercise link", business_id=business.id, ru_ref=business.ru_ref)
-            business.instruments.remove(instrument)
+        if instrument.type == "SEFT":
+            # SEFT instruments need to be deleted from both GCP and the db, removing just the link will leave orphaned
+            # data. When deleting SEFT instruments, the link will automatically be removed without this function
+            raise RasError(
+                f"{instrument_id} is of type SEFT which should be deleted and not unlinked",
+                405,
```

Comment:
Debatable on the 405

Reply:
405 makes sense, do we want the error (depending how often this is attempted against something not valid) if thats not an issue then fine

Reply:
It should feasibly never happen as it would need someone to either hit the end point or mess around with the UI, which is highly unlikely considering it's internal. I think it's still good pratice to do that, but I can see it's not a pattern we do much in RAS/RM. One for tech maybe, as well as how we handle api calls between services